heal/vlmd/validate/validate.py

```diff
@@ -100,32 +124,43 @@ def vlmd_validate(
             message = "Could not read csv data from input"
             logger.error(message)
             raise ValidationError(message)
-    elif file_suffix == "json":
+    elif file_suffix == "json" and data is None:
         logger.debug("Getting json data from file")
         data = read_data_from_json_file(input_file)
 
-    # if json then try a validation
+    # if input is json then try a validation and return input
     if file_suffix == "json":
         logger.debug("Validating json data")
         try:
             jsonschema.validate(instance=data, schema=schema)
-        except jsonschema.ValidationError as e:
+        except jsonschema.ValidationError as err:
             logger.error("Error in validating json input")
-            raise e
-
-    # convert
-    input_type = file_type_to_fxn_map.get(file_suffix)
-    if not input_type:
+            raise err
+        logger.debug("JSON input is valid")
+        if return_converted_output:
+            return data
+        else:
+            return True
+
+    # else non-json data should first be converted and then validated
+    if file_type == "auto":
+        logger.debug(f"Using file_suffix for file_type {file_suffix}")
+        file_convert_function = file_type_to_fxn_map.get(file_suffix)
+    else:
+        file_convert_function = file_type_to_fxn_map.get(file_type)
+    if not file_convert_function:
         message = f"Could not get conversion function from file_suffix '{file_suffix}'"
         logger.error(message)
         raise ExtractionError(message)
     data_dictionaries = {}
-    logger.debug(f"Verifying vlmd can be converted using input_type '{input_type}'")
+    logger.debug(
+        f"Verifying vlmd can be converted using file conversion function '{file_convert_function}'"
+    )
     data_dictionary_props = {}
     try:
         data_dictionaries = convert_to_vlmd(
             input_filepath=input_file,
-            input_type=input_type,
+            input_type=file_convert_function,
```

Comment:
If I'm reading this right, there is a edge case: if the input VLMD is a CSV file but has no `standardsMapping[0].instrument.title` column, the validation would fail because there will be no `title` in the converted VLMD JSON? Although when you are running `vlmd extract` you should be able to add a title via the arguments, but the `vlmd validate` here does not know that?

Reply:
The validation adds a dummy title to CSV input. Good CSV data will pass validation without a user-supplied title.